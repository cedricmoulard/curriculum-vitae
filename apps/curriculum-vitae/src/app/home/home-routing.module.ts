import { NgModule } from "@angular/core";
import { RouterModule, Routes } from "@angular/router";
import { HomeComponent } from "./home.component";
import { ExperiencesComponent } from "../experiences/experiences.component";

const routes: Routes = [
  { path: "", component: HomeComponent, data: { animation: "HomePage" } }
];

@NgModule({
  imports: [RouterModule.forChild(routes)],
  exports: [RouterModule]
})
export class HomeRoutingModule {}
