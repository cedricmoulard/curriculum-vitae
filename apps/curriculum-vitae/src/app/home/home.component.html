<div class="container">
  <div class="title">
    {{ title }}
  </div>
  <div class="subtitle">ScrumMaster | Tech Lead Full Stack</div>
</div>
