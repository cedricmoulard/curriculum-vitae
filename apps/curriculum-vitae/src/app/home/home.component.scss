.container {
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;

  height: 100vh;

  .title {
    font-size: 1.3em;
  }

  .subtitle {
  }
}
