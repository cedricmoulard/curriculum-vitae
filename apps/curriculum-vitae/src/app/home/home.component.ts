import { Component, OnInit } from "@angular/core";
import {
  animate,
  state,
  style,
  transition,
  trigger
} from "@angular/animations";
import { Router } from "@angular/router";

@Component({
  selector: "cv-home",
  templateUrl: "./home.component.html",
  styleUrls: ["./home.component.scss"],
  animations: [
    trigger("titleAnimation", [
      state(
        "true",
        style({
          transform: "scale(1.2)",
          opacity: 1
        })
      ),
      transition(":enter", animate("1000ms ease-out"))
    ])
  ]
})
export class HomeComponent {
  title = "CÃ©dric Moulard";
  subtitle = "ScrumMaster | Tech Lead Full Stack";

  constructor(private readonly router: Router) {}

  goToExperiences() {
    this.router.navigate(["/experiences"]);
  }
}
