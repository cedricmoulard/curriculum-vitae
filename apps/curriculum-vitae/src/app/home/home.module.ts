import { NgModule } from "@angular/core";
import { CommonModule } from "@angular/common";
import { HomeComponent } from "./home.component";
import { HomeRoutingModule } from "./home-routing.module";
import { MatButtonModule } from "@angular/material";

@NgModule({
  declarations: [HomeComponent],
  imports: [CommonModule, HomeRoutingModule, MatButtonModule],
  exports: [HomeComponent]
})
export class HomeModule {}
