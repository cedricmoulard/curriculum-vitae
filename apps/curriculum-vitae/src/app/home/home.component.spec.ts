import { async, ComponentFixture, TestBed } from "@angular/core/testing";

import { HomeComponent } from "./home.component";
import { RouterTestingModule } from "@angular/router/testing";
import { BrowserAnimationsModule } from "@angular/platform-browser/animations";

describe("HomeComponent", () => {
  let component: HomeComponent;
  let fixture: ComponentFixture<HomeComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [HomeComponent],
      imports: [BrowserAnimationsModule, RouterTestingModule],
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(HomeComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it("should create", () => {
    expect(component).toBeTruthy();
  });

  it(`should have as title 'Cédric Moulard'`, () => {
    expect(component.title).toEqual("Cédric Moulard");
  });

  it("should render match snapshot", () => {
    const compiled = fixture.debugElement.nativeElement;
    expect(compiled).toMatchSnapshot();
  });
});
