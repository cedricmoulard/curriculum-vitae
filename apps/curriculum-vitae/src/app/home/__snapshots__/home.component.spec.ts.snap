// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomeComponent should render match snapshot 1`] = `
<div
  id="root2"
  ng-version="8.0.0"
>
  <div
    class="container"
  >
    <div
      class="title"
    >
       CÃ©dric Moulard 
    </div>
    <div
      class="subtitle"
    >
      ScrumMaster | Tech Lead Full Stack
    </div>
  </div>
</div>
`;
